code/client/makecatalogs

```diff
@@ -150,6 +199,29 @@ def makecatalogs(repopath, options):
                 if key.startswith('_'):
                     del(pkginfo[key])
 
+
+            if iconhashing:
+
+                name = pkginfo.get('name')
+
+                if pkginfo.get('icon_name'):
+                    try:
+                        iconhash = icons.get(icon_name)
```

Comment:
`icon_name` is undefined - so this would raise an exception, but maybe not for the reason you want..

If it _were_ defined, the try/except and the one below (line 214) are doing a try on something that's unlikely to ever raise an exception, since `icons.get()` will just return `None` here on a missing key. I think it's more idiomatic to just say `icons.get(thething)` and then do what you want based on whether you got a not-None value - or in the case below, you're later checking if `iconhash` is `not None` anyway.